eclipse-che/che-server
# feat: Allow to configure 2 gitlab providers simultaneously

### What does this PR do?
Allow to configure 2 gitlab providers simultaneously

### Screenshot/screencast of this PR

![screenshot-cluster-admin-che-dashboard-local-server_apps_rosa_oaky8-tf8zx-r5p_asdn_p3_openshiftapps_com-2024_10_23-11_53_49](https://github.com/user-attachments/assets/2a88f1eb-70ba-4006-be0c-8dd2aff8746c)
![screenshot-cluster-admin-che-dashboard-local-server_apps_rosa_oaky8-tf8zx-r5p_asdn_p3_openshiftapps_com-2024_10_23-11_54_23](https://github.com/user-attachments/assets/01fc8088-b6a8-4126-818d-72e5e1a3c4b7)


### What issues does this PR fix or reference?

https://issues.redhat.com/browse/CRW-7142

### How to test this PR?

1. Deploy `che-server` image: `quay.io/eclipse/che-server:pr-731` 
2. Deploy `che-operator` image: `quay.io/ivinokur/che-operator:next`
3. Deploy `che-dashboard` image: `quay.io/eclipse/che-dashboard:pr-1237`
4. [Configure oauth](https://eclipse.dev/che/docs/stable/administration-guide/configuring-oauth-2-for-gitlab/) for two gitlab instances, add a random suffix for the second oauth secret name.
5. Start a workspace from each gitlab repository with a devfille.
6. See: personal access token is generated for both gitlab instances.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Release Notes
With this feature it is possible to configure two Gitlab oauth providers simultaneously.
### Reviewers

Reviewers, please comment how you tested the PR when approving it.
